src/gda/backend_gda.cpp

```diff
@@ -673,13 +677,26 @@ void GDABackend::autodetect_dv_libs() {
       DPRINTF("Initializing rocSHMEM MLX5 GDA support failed\n");
     }
   }
+#endif // defined(GDA_MLX5)
 
   if (gda_vendor == GDAVendor::NONE) {
-    printf("Initializing rocSHMEM with IONIC, BNXT, or MLX5 GDA support failed\n");
+    DPRINTF("Initializing rocSHMEM with IONIC, BNXT, or MLX5 GDA support failed: no DV library found\n");
     abort();
   }
 }
 
+void GDABackend::close_dv_libs() {
+  if (ionicdv_handle_ != nullptr)
+    dlclose(ionicdv_handle_);
+
+  if (bnxtdv_handle_ != nullptr)
+    dlclose(bnxtdv_handle_);
+
+  if (mlx5dv_handle_ != nullptr)
+    dlclose(mlx5dv_handle_);
```

Comment:
```suggestion
  if (ionicdv_handle_) {
    dlclose(ionicdv_handle_);
  }

  if (bnxtdv_handle_) {
    dlclose(bnxtdv_handle_);
  }

  if (mlx5dv_handle_) {
    dlclose(mlx5dv_handle_);
  }
```

Comparison with `nullptr` is redundant, add braces to avoid future errors e.g. `goto fail`.